src/test/java/de/tum/in/www1/artemis/architecture/ResourceArchitectureTest.java

```diff
@@ -69,6 +73,32 @@ void shouldCorrectlyUseRequestMappingAnnotations() {
         }
     }
 
+    @Test
+    void testUseKebabCaseForRestEndpoints() {
+        for (var annotation : annotationClasses) {
+            methods().should(useKebabCaseForRestAnnotations(annotation)).check(productionClasses);
+        }
+    }
+
+    protected ArchCondition<JavaMethod> useKebabCaseForRestAnnotations(Class<?> annotationClass) {
+        return new ArchCondition<>("use kebab case for rest mapping annotations") {
+
+            @Override
+            public void check(JavaMethod item, ConditionEvents events) {
+                var restMappingAnnotation = item.getAnnotations().stream()
+                        .filter(annotation -> ((JavaClass) annotation.getType()).getSimpleName().equals(annotationClass.getSimpleName())).findFirst();
+
+                if (restMappingAnnotation.isPresent()) {
+                    boolean satisfied = Arrays.stream(((String[]) restMappingAnnotation.get().tryGetExplicitlyDeclaredProperty("value").get())[0].split("/"))
```

Comment:
Why don't you match the complete path?

Reply:
no idea why I didn't think of that. I made it into one regex, which is quite huge, but the code using it is simpler.